src/pudl/metadata/resources/mcoe.py

```diff
@@ -0,0 +1,224 @@
+                "factor is calculated based on reported generator capacity and the "
+                "allocated net generation reported in the generation and generation "
+                "fuel tables."
+            ),
+            "schema": {
+                "fields": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                    "net_generation_mwh",
+                    "capacity_mw",
+                    "capacity_factor",
+                ],
+                "primary_key": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                ],
+            },
+            "field_namespace": "eia",
+            "sources": ["eia923", "eia860"],
+            "etl_group": "outputs",
+        }
+        for freq in AGG_FREQS
+    }
+    | {
+        f"fuel_cost_by_generator_{freq}": {
+            "description": (
+                f"{freq.title()} estimate of per-generator fuel costs both per MMBTU "
+                "and per MWh. These calculations are based on the allocation of net "
+                "generation and plant-level delivered fuel prices reported in the fuel "
+                "receipts and cost table, along with the un-allocated fuel consumption "
+                "from the boiler fuel table. This means fuel cost coverage is low."
+                "The fuel costs are also currently aggregated to coarse fuel "
+                "categories rather than using the more detailed energy source codes."
+            ),
+            "schema": {
+                "fields": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                    "unit_id_pudl",
+                    "plant_name_eia",
+                    "plant_id_pudl",
+                    "utility_id_eia",
+                    "utility_name_eia",
+                    "utility_id_pudl",
+                    "fuel_type_count",
+                    "fuel_type_code_pudl",
+                    "fuel_cost_from_eiaapi",
+                    "fuel_cost_per_mmbtu",
+                    "heat_rate_mmbtu_mwh",
+                    "fuel_cost_per_mwh",
+                ],
+                "primary_key": [
+                    "report_date",
+                    "plant_id_eia",
+                    "generator_id",
+                ],
+            },
+            "field_namespace": "eia",
+            "sources": ["eia923", "eia860"],
+            "etl_group": "outputs",
+        }
+        for freq in AGG_FREQS
+    }
+    | {
+        f"mcoe_{freq}": {
+            "description": (
+                f"{freq.title()} generator capacity factor, heat rate, fuel cost per MMBTU and fuel cost "
+                "per MWh. These calculations are based on the allocation of net generation reported on "
+                "the basis of plant, prime mover and energy source to individual generators. Heat rates "
```

Comment:
I updated this description to reflect the new way for estimating heat rates using allocated estimates for net generation and fuel consumption. Does this seem right?

Reply:
I think we should probably also note that there are good reasons why one might get ridiculous numbers for some of these outputs, and that those outputs have been filtered out of these results.

However, doing this filtering only at the final step means that there are probably some accidentally reasonable numbers that get left in the outputs.

Reply:
I updated the descriptions for the intermediate tables where ridiculous numbers are expected and additionally added a helper function (`oob_to_nan_with_dependent_cols`) to filter out accidentally reasonable calculations.